### PR TITLE
fix: ui and ux improvements in booking questions

### LINKED ISSUE #7482: Support asterisk for required fields
https://github.com/calcom/cal.com/blob/517cfde5b8a0e4910ea27f21445a516ea372a3ce/packages/features/form-builder/Components.tsx#L326

```javascript

import { useEffect } from "react";
import type { z } from "zod";

import type {
  TextLikeComponentProps,
  SelectLikeComponentProps,
} from "@calcom/app-store/ee/routing-forms/components/react-awesome-query-builder/widgets";
import Widgets from "@calcom/app-store/ee/routing-forms/components/react-awesome-query-builder/widgets";
import { classNames } from "@calcom/lib";
import { useLocale } from "@calcom/lib/hooks/useLocale";
import type { BookingFieldType } from "@calcom/prisma/zod-utils";
import { PhoneInput, AddressInput, Button, Label, Group, RadioField, EmailField, Tooltip } from "@calcom/ui";
import { FiUserPlus, FiX } from "@calcom/ui/components/icon";

import { ComponentForField } from "./FormBuilder";
import type { fieldsSchema } from "./FormBuilderFieldsSchema";

type Component =
  | {
      propsType: "text";
      factory: <TProps extends TextLikeComponentProps>(props: TProps) => JSX.Element;
    }
  | {
      propsType: "textList";
      factory: <TProps extends TextLikeComponentProps<string[]>>(props: TProps) => JSX.Element;
    }
  | {
      propsType: "select";
      factory: <TProps extends SelectLikeComponentProps>(props: TProps) => JSX.Element;
    }
  | {
      propsType: "boolean";
      factory: <TProps extends TextLikeComponentProps<boolean>>(props: TProps) => JSX.Element;
    }
  | {
      propsType: "multiselect";
      factory: <TProps extends SelectLikeComponentProps<string[]>>(props: TProps) => JSX.Element;
    }
  | {
      // Objective type question with option having a possible input
      propsType: "objectiveWithInput";
      factory: <
        TProps extends SelectLikeComponentProps<{
          value: string;
          optionValue: string;
        }> & {
          optionsInputs: NonNullable<z.infer<typeof fieldsSchema>[number]["optionsInputs"]>;
          value: { value: string; optionValue: string };
        } & {
          name?: string;
        }
      >(
        props: TProps
      ) => JSX.Element;
    };

// TODO: Share FormBuilder components across react-query-awesome-builder(for Routing Forms) widgets.
// There are certain differences b/w two. Routing Forms expect label to be provided by the widget itself and FormBuilder adds label itself and expect no label to be added by component.
// Routing Form approach is better as it provides more flexibility to show the label in complex components. But that can't be done right now because labels are missing consistent asterisk required support across different components
export const Components: Record<BookingFieldType, Component> = {
  text: {
    propsType: "text",
    factory: (props) => <Widgets.TextWidget noLabel={true} {...props} />,
  },
  textarea: {
    propsType: "text",
    // TODO: Make rows configurable in the form builder
    factory: (props) => <Widgets.TextAreaWidget rows={3} {...props} />,
  },
  number: {
    propsType: "text",
    factory: (props) => <Widgets.NumberWidget noLabel={true} {...props} />,
  },
  name: {
    propsType: "text",
    // Keep special "name" type field and later build split(FirstName and LastName) variant of it.
    factory: (props) => <Widgets.TextWidget noLabel={true} {...props} />,
  },
  phone: {
    propsType: "text",
    factory: ({ setValue, readOnly, ...props }) => {
      if (!props) {
        return <div />;
      }

      return (
        <PhoneInput
          disabled={readOnly}
          onChange={(val: string) => {
            setValue(val);
          }}
          {...props}
        />
      );
    },
  },
  email: {
    propsType: "text",
    factory: (props) => {
      if (!props) {
        return <div />;
      }
      return <Widgets.TextWidget type="email" noLabel={true} {...props} />;
    },
  },
  address: {
    propsType: "text",
    factory: (props) => {
      return (
        <AddressInput
          onChange={(val) => {
            props.setValue(val);
          }}
          {...props}
        />
      );
    },
  },
  multiemail: {
    propsType: "textList",
    //TODO: Make it a ui component
    factory: function MultiEmail({ value, readOnly, label, setValue, ...props }) {
      const placeholder = props.placeholder;
      const { t } = useLocale();
      value = value || [];
      const inputClassName =
        "dark:placeholder:text-darkgray-600 focus:border-brand dark:border-darkgray-300 dark:text-darkgray-900 block w-full rounded-md border-gray-300 text-sm focus:ring-black disabled:bg-gray-200 disabled:hover:cursor-not-allowed dark:bg-transparent dark:selection:bg-green-500 disabled:dark:text-gray-500";
      return (
        <>
          {value.length ? (
            <div>
              <label
                htmlFor="guests"
                className="mb-1 block text-sm font-medium text-gray-700 dark:text-white">
                {label}
              </label>
              <ul>
                {value.map((field, index) => (
                  <li key={index}>
                    <EmailField
                      disabled={readOnly}
                      value={value[index]}
                      onChange={(e) => {
                        value[index] = e.target.value;
                        setValue(value);
                      }}
                      className={classNames(inputClassName, "border-r-0")}
                      addOnClassname={classNames(
                        "border-gray-300 border block border-l-0 disabled:bg-gray-200 disabled:hover:cursor-not-allowed bg-transparent disabled:text-gray-500 dark:border-darkgray-300 "
                      )}
                      placeholder={placeholder}
                      label={<></>}
                      required
                      addOnSuffix={
                        !readOnly ? (
                          <Tooltip content="Remove email">
                            <button
                              className="m-1 disabled:hover:cursor-not-allowed"
                              type="button"
                              onClick={() => {
                                value.splice(index, 1);
                                setValue(value);
                              }}>
                              <FiX className="text-gray-600" />
                            </button>
                          </Tooltip>
                        ) : null
                      }
                    />
                  </li>
                ))}
              </ul>
              {!readOnly && (
                <Button
                  type="button"
                  color="minimal"
                  StartIcon={FiUserPlus}
                  className="my-2.5"
                  onClick={() => {
                    value.push("");
                    setValue(value);
                  }}>
                  {t("add_another")}
                </Button>
              )}
            </div>
          ) : (
            <></>
          )}

          {!value.length && !readOnly && (
            <Button
              color="minimal"
              variant="button"
              StartIcon={FiUserPlus}
              onClick={() => {
                value.push("");
                setValue(value);
              }}
              className="mr-auto">
              {label}
            </Button>
          )}
        </>
      );
    },
  },
  multiselect: {
    propsType: "multiselect",
    factory: (props) => {
      const newProps = {
        ...props,
        listValues: props.options.map((o) => ({ title: o.label, value: o.value })),
      };
      return <Widgets.MultiSelectWidget {...newProps} />;
    },
  },
  select: {
    propsType: "select",
    factory: (props) => {
      const newProps = {
        ...props,
        listValues: props.options.map((o) => ({ title: o.label, value: o.value })),
      };
      return <Widgets.SelectWidget {...newProps} />;
    },
  },
  checkbox: {
    propsType: "multiselect",
    factory: ({ options, readOnly, setValue, value }) => {
      value = value || [];
      return (
        <div>
          {options.map((option, i) => {
            return (
              <label key={i} className="block">
                <input
                  type="checkbox"
                  disabled={readOnly}
                  onChange={(e) => {
                    const newValue = value.filter((v) => v !== option.value);
                    if (e.target.checked) {
                      newValue.push(option.value);
                    }
                    setValue(newValue);
                  }}
                  className="dark:bg-darkgray-300 dark:border-darkgray-300 h-4 w-4 rounded border-gray-300 text-black focus:ring-black ltr:mr-2 rtl:ml-2"
                  value={option.value}
                  checked={value.includes(option.value)}
                />
                <span className="text-sm ltr:ml-2 ltr:mr-2 rtl:ml-2 dark:text-white">
                  {option.label ?? ""}
                </span>
              </label>
            );
          })}
        </div>
      );
    },
  },
  radio: {
    propsType: "select",
    factory: ({ setValue, value, options }) => {
      return (
        <Group
          value={value}
          onValueChange={(e) => {
            setValue(e);
          }}>
          <>
            {options.map((option, i) => (
              <RadioField
                label={option.label}
                key={`option.${i}.radio`}
                value={option.label}
                id={`option.${i}.radio`}
              />
            ))}
          </>
        </Group>
      );
    },
  },
  radioInput: {
    propsType: "objectiveWithInput",
    factory: function RadioInputWithLabel({ name, options, optionsInputs, value, setValue, readOnly }) {
      useEffect(() => {
        if (!value) {
          setValue({
            value: options[0]?.value,
            optionValue: "",
          });
        }
      }, [options, setValue, value]);

      return (
        <div>
          <div>
            <div className="mb-2">
              {options.length > 1 ? (
                options.map((option, i) => {
                  return (
                    <label key={i} className="block">
                      <input
                        type="radio"
                        disabled={readOnly}
                        name={name}
                        className="dark:bg-darkgray-300 dark:border-darkgray-300 h-4 w-4 border-gray-300 text-black focus:ring-black ltr:mr-2 rtl:ml-2"
                        value={option.value}
                        onChange={(e) => {
                          setValue({
                            value: e.target.value,
                            optionValue: "",
                          });
                        }}
                        checked={value?.value === option.value}
                      />
                      <span className="text-sm ltr:ml-2 ltr:mr-2 rtl:ml-2 dark:text-white">
                        {option.label ?? ""}
                      </span>
                    </label>
                  );
                })
              ) : (
                // Show option itself as label because there is just one option
                // TODO: Support asterisk for required fields
                <Label>{options[0].label}</Label>
              )}
            </div>
          </div>
          {(() => {
            const optionField = optionsInputs[value?.value];
            if (!optionField) {
              return null;
            }
            return (
              <div>
                <ComponentForField
                  readOnly={!!readOnly}
                  field={{
                    ...optionField,
                    name: "optionField",
                  }}
                  value={value?.optionValue}
                  setValue={(val: string) => {
                    setValue({
                      value: value?.value,
                      optionValue: val,
                    });
                  }}
                />
              </div>
            );
          })()}
        </div>
      );
    },
  },
  boolean: {
    propsType: "boolean",
    factory: ({ readOnly, label, value, setValue }) => {
      return (
        <div className="flex">
          <input
            type="checkbox"
            onChange={(e) => {
              if (e.target.checked) {
                setValue(true);
              } else {
                setValue(false);
              }
            }}
            className="h-4 w-4 rounded border-gray-300 text-black focus:ring-black disabled:bg-gray-200 ltr:mr-2 rtl:ml-2 disabled:dark:text-gray-500"
            placeholder=""
            checked={value}
            disabled={readOnly}
          />
          <Label className="-mt-px block text-sm font-medium text-gray-700 dark:text-white">{label}</Label>
        </div>
      );
    },
  },
} as const;
// Should use `statisfies` to check if the `type` is from supported types. But satisfies doesn't work with Next.js config

```

### LINKED ISSUE #7541: [CAL-1187] Good to have Improvements in Booking Questions
1. Location field is confusing because it is marked as visible question and we cannot hide it by toggling but it is hidden when user has only one location in event type so it does not really shows up in the form every time ( like additional notes).

1. 

2\.1) we should also use \* to denote required fields. Like placeholder is an optional field here

2\.2) disable "required" field ( the last field below) when we are not allowed to make changes like selecting yes or no. For example when editing Name field we cannot mark this question as not required by selecting no here. we should also display not allowed when cursor hovers over it.

3. maybe in later releases we can also show how the input would look like as user is changing different properties like name, label like storybook

[https://user-images.githubusercontent.com/53316345/223189875-dfe9acb9-1e9d-4944-af10-0b08cf782c11.mov](https://user-images.githubusercontent.com/53316345/223189875-dfe9acb9-1e9d-4944-af10-0b08cf782c11.mov)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1187](https://linear.app/calcom/issue/CAL-1187/good-to-have-improvements-in-booking-questions)</sub>

### LINKED ISSUE #7552: [CAL-1333] No * at phone number and location, although required.
Nice to see you have now included the * in the name and email address. Just noticed that it is missing from the phone number and location though.

The location can also not be set to required in the settings.

Bildschirm­foto 2023-03-07 um 03 59 39
Bildschirm­foto 2023-03-07 um 03 59 29

<sub>[CAL-1333](https://linear.app/calcom/issue/CAL-1333/no-at-phone-number-and-location-although-required)</sub>

### LINKED ISSUE #7704: [CAL-1242] public booking page: Asterix position breaking with 2 lines
Looks like it’s floating out and absolutely positioned. Should be after the last character like the first line.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1242](https://linear.app/calcom/issue/CAL-1242/public-booking-page-asterix-position-breaking-with-2-lines)</sub>

### LINKED ISSUE #7761: [CAL-1271] Missing asterisk for phone number in booking form
An asterisk is missing at the end to mark the mandatory information. "Telefonnummer Teilnehmer*in" should also be renamed to "Telefonnummer".

Bildschirm­foto 2023-03-16 um 01 05 14

<sub>[CAL-1271](https://linear.app/calcom/issue/CAL-1271/missing-asterisk-for-phone-number-in-booking-form)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
Fixes https://github.com/calcom/cal.com/issues/7541
Fixes https://github.com/calcom/cal.com/issues/7552
Fixes https://github.com/calcom/cal.com/issues/7482
Fixes https://github.com/calcom/cal.com/issues/7704

Refer to the discussion and loom videos here https://threads.com/34453537469?s=9nKsam7Qo9LfMJrL8JrYmU

- [x] Don't show location field when less than two options
- [x] fix missing translations
- [x] Hide not required things like toggle switch, delete icon
- [x] add padding, hover and tooltip in switch
- [x] remove hover from badge and few more ui issues
- [x] Show asterisk in location field
- [x] Fix tests
- [ ] Make changes recommended by Ciaran in dropdown of input type. (will fix this in another PR)
- [x] Fix asterisk postion in multiline lable

https://user-images.githubusercontent.com/53316345/225123834-329a86a8-5116-4554-9043-d8e5f99130c4.mov

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)

### CODE REVIEW

File: apps/web/components/eventtype/EventAdvancedTab.tsx [hunk 1]
```diff
@@ -147,6 +148,7 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupProps,
         description={t("booking_questions_description")}
         addFieldLabel={t("add_a_booking_question")}
         formProp="bookingFields"
+        eventTypeLocationsLength={eventTypeLocationsLength}
```
Comment (zomars) [inline apps/web/components/eventtype/EventAdvancedTab.tsx:151]: Can we save FormBuilder to having to know more than needed.

```suggestion
        shouldDisplayLocation ={eventType.locations.length > 1}
```

File: packages/features/form-builder/FormBuilder.tsx [hunk 2]
```diff
@@ -44,11 +44,13 @@ export const FormBuilder = function FormBuilder({
   description,
   addFieldLabel,
   formProp,
+  eventTypeLocationsLength,
 }: {
   formProp: string;
   title: string;
   description: string;
   addFieldLabel: string;
+  eventTypeLocationsLength: number;
```
Comment (zomars) [inline packages/features/form-builder/FormBuilder.tsx:53]: Why not?

```suggestion
  shouldDisplayLocation: boolean;
```

File: packages/features/form-builder/FormBuilder.tsx [hunk 3]
```diff
@@ -282,14 +287,19 @@ export const FormBuilder = function FormBuilder({
               return groupBy;
             }, {} as Record<string, NonNullable<(typeof field)["sources"]>>);
 
+            // Only show location field when it requires selection of input from the booker
+            if (field.name === "location" && eventTypeLocationsLength < 2) {
```
Comment (zomars) [inline packages/features/form-builder/FormBuilder.tsx:291]:
```suggestion
            if (field.name === "location" && !shouldDisplayLocation) {
```

File: packages/features/form-builder/FormBuilder.tsx [hunk 4]
```diff
@@ -319,42 +329,42 @@ export const FormBuilder = function FormBuilder({
...
+                        disabled={isFieldEditableSystem}
+                        checked={!field.hidden}
+                        onCheckedChange={(checked) => {
+                          update(index, { ...field, hidden: !checked });
+                        }}
+                        switchContainerClassName="p-2 hover:bg-gray-100 rounded"
+                        tooltip={t("show_on_booking_page")}
+                      />
+                    )}
+                    {!isFieldEditableSystem && !isFieldEditableSystemButOptional && (
```
Comment (zomars) [inline packages/features/form-builder/FormBuilder.tsx:350]:
```suggestion
                    {shouldDisplayDelete && (
```

File: packages/features/form-builder/FormBuilder.tsx [hunk 5]
```diff
@@ -267,6 +270,8 @@ export const FormBuilder = function FormBuilder({
             const fieldType = FieldTypesMap[field.type];
 
             const isRequired = field.required;
+            const isFieldEditableSystemButOptional = field.editable === "system-but-optional";
+            const isFieldEditableSystem = field.editable === "system";
```
Comment (zomars) [inline packages/features/form-builder/FormBuilder.tsx:274]:
```suggestion
            const isFieldEditableSystem = field.editable === "system";
            const shouldDisplayDelete = !isFieldEditableSystem && !isFieldEditableSystemButOptional;
```

File: packages/ui/components/form/switch/Switch.tsx [hunk 6]
```diff
@@ -21,14 +21,20 @@ const Switch = (
     };
     fitToHeight?: boolean;
     tooltip?: string;
+    switchContainerClassName?: string;
```
Comment (zomars) [inline packages/ui/components/form/switch/Switch.tsx:24]:
```suggestion
```

Let's update the whole prop signature to this:

```
  props: React.ComponentProps<typeof PrimitiveSwitch.Root> & {
    label?: string;
    classNames?: {
      container?: string;
      thumb?: string;
    };
    fitToHeight?: boolean;
    tooltip?: string;
  }
```

**Review by zomars — CHANGES_REQUESTED** [review]
Nice work @Udit-takkar left some nitpicks for code. 🙏🏽

File: packages/features/form-builder/FormBuilder.tsx [hunk 7]
```diff
@@ -718,6 +729,19 @@ export const FormBuilderField = ({
...
+                    return (
+                      <div
+                        data-testid="error-message-location"
+                        className="mt-2 flex items-center text-sm text-red-700 ">
+                        <FiInfo className="h-3 w-3 ltr:mr-2 rtl:ml-2" />
+                        <p>{t("error_required_field")}</p>
+                      </div>
+                    );
+                  }
+
```
Comment (Udit-takkar) [inline packages/features/form-builder/FormBuilder.tsx:744]: @hariombalhara is there any better way to show this required error for location field ? location is marked as optional in zod schema.

**Comment by PeerRich** [comment]
@zomars should NIT's be blocking? right now requesting changes blocks the PR

**Comment by zomars** [comment]
> @zomars should NIT's be blocking? right now requesting changes blocks the PR

Maybe I've used the word nitpick wrong. But I'm requesting changes before merging. So it should be blocking.

File: packages/features/form-builder/FormBuilder.tsx [hunk 8]
```diff
@@ -44,11 +44,13 @@ export const FormBuilder = function FormBuilder({
   description,
   addFieldLabel,
   formProp,
+  shouldDisplayLocation,
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilder.tsx:47]: FormBuilder can't have hardcoded info for a field. It is a general purpose component and it is supposed to power Routing Forms as well.

Comment (hariombalhara) [reply packages/features/form-builder/FormBuilder.tsx:47]: We can possibly solve it by supporting a field like `hideWhenJustOneOption` and setting it to true for location field.

File: packages/features/form-builder/FormBuilder.tsx (hunk 7 above)
Comment (hariombalhara) [reply packages/features/form-builder/FormBuilder.tsx:746]: I don't really understand why this is required.  On production, if just Attendee Phone Number is chosen as location, it is considered required.
Screenshot 2023-03-15 at 11 12 27 PM

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Left some blocking comments.

File: packages/features/form-builder/FormBuilder.tsx (hunk 7 above)
Comment (Udit-takkar) [reply packages/features/form-builder/FormBuilder.tsx:746]: @hariombalhara  if you completely clear attendee phone number even +91 then you won't see this error. ( this is very small edge case)

File: packages/features/form-builder/FormBuilder.tsx (hunk 8 above)
Comment (Udit-takkar) [reply packages/features/form-builder/FormBuilder.tsx]: okay i added this field hideWhenJustOneOption but now if i add more location and click save and go to advanced tab. i would have to reload to make sure location field is displayed.

File: packages/features/form-builder/FormBuilder.tsx (hunk 7 above)
Comment (hariombalhara) [reply packages/features/form-builder/FormBuilder.tsx]: Thanks @Udit-takkar. Didn't notice that was happening. 
Have fixed it in a better way now.

File: packages/features/bookings/lib/getBookingResponsesSchema.ts [hunk 9]
```diff
@@ -98,12 +98,12 @@ function preprocess<T extends z.ZodType>({
         const views = bookingField.views;
         const isFieldApplicableToCurrentView =
           currentView === "ALL_VIEWS" ? true : views ? views.find((view) => view.id === currentView) : true;
+        let hidden = bookingField.hidden;
+        if (bookingField.hideWhenJustOneOption) {
+          hidden = hidden || bookingField.options?.length === 1;
```
Comment (Udit-takkar) [inline packages/features/bookings/lib/getBookingResponsesSchema.ts:103]: @hariombalhara  should be <=1 as some event types could have empty array as location.

Comment (hariombalhara) [reply packages/features/bookings/lib/getBookingResponsesSchema.ts:103]: I thought about this. A field with no options shouldn't be allowed to be created ideally. But we can still improve this check like this probably

File: packages/features/form-builder/FormBuilder.tsx [hunk 10]
```diff
@@ -444,7 +463,7 @@ export const FormBuilder = function FormBuilder({
                   fieldForm.getValues("editable") === "system" ||
                   fieldForm.getValues("editable") === "system-but-optional"
                 }
-                label="Name"
+                label={t("name")}
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilder.tsx:466]:
```suggestion
                label="Identifier"
```

Comment (Udit-takkar) [reply packages/features/form-builder/FormBuilder.tsx]: @hariombalhara no i18n here ?

File: packages/features/bookings/lib/getLocationOptionsForSelect.ts
Comment (zomars) [inline packages/features/bookings/lib/getLocationOptionsForSelect.ts:1]: https://handbook.cal.com/engineering/best-practices/dont-modularize-prematurely

File: packages/features/bookings/lib/getBookingResponsesSchema.ts [hunk 11]
```diff
@@ -98,12 +98,13 @@ function preprocess<T extends z.ZodType>({
         const views = bookingField.views;
         const isFieldApplicableToCurrentView =
           currentView === "ALL_VIEWS" ? true : views ? views.find((view) => view.id === currentView) : true;
+        let hidden = bookingField.hidden;
+        const numOptions = bookingField.options?.length ?? 0;
+        if (bookingField.hideWhenJustOneOption) {
+          hidden = hidden || numOptions <= 1;
+        }
```
Comment (zomars) [inline packages/features/bookings/lib/getBookingResponsesSchema.ts:105]:
```suggestion
        const numOptions = bookingField.options?.length ?? 0;
        const isRequired = !bookingField.hidden && isFieldApplicableToCurrentView && bookingField.required;
        if (bookingField.hideWhenJustOneOption && numOptions <= 1) {
          isRequired = false;
        }
```

File: apps/web/components/eventtype/EventAdvancedTab.tsx [hunk 12]
```diff
@@ -8,6 +8,7 @@ import { v5 as uuidv5 } from "uuid";
 
 import type { EventNameObjectType } from "@calcom/core/event";
 import { getEventName } from "@calcom/core/event";
+import getLocationsOptionsForSelect from "@calcom/features/bookings/lib/getLocationOptionsForSelect";
```
Comment (zomars) [inline apps/web/components/eventtype/EventAdvancedTab.tsx:11]: https://handbook.cal.com/engineering/best-practices/dont-modularize-prematurely

**Review by zomars — CHANGES_REQUESTED** [review]
Requesting more change 🙏🏽

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 13]
```diff
@@ -30,6 +29,7 @@ import {
 import getBookingResponsesSchema, {
   getBookingResponsesPartialSchema,
 } from "@calcom/features/bookings/lib/getBookingResponsesSchema";
+import getLocationOptionsForSelect from "@calcom/features/bookings/lib/getLocationOptionsForSelect";
```
[the remark this answers is not in the document]
Comment (hariombalhara) [reply apps/web/components/booking/pages/BookingPage.tsx:32]: It's being used in BookingPage.tsx as well as EventAdvancedTab.tsx. So, I think that rule isn't applicable here.

File: packages/features/bookings/lib/getBookingResponsesSchema.ts (hunk 11 above)
Comment (hariombalhara) [reply packages/features/bookings/lib/getBookingResponsesSchema.ts:105]: I find the current approach better to understand logically.
Whether a field is hidden or not is decided either by `bookingField.hidden`(which is explicitly decided by the user) or decided automatically by Cal if that field has one option(assuming hideWhenJustOneOption is true)

Now, whether a field is required or not is decided by 3 factors(
- It shouldn't be hidden
- It shouldn't be non-applicable to the current view
- It should be required.

File: apps/web/public/static/locales/en/common.json [hunk 14]
```diff
@@ -1627,6 +1627,7 @@
   "form_builder_field_add_subtitle": "Customize the questions asked on the booking page",
   "form_builder_system_field_cant_delete": "This system field can't be removed.",
   "form_builder_system_field_cant_toggle": "This system field can't be toggled.",
```
Comment (hariombalhara) [inline apps/web/public/static/locales/en/common.json:1629]:
```suggestion
```

File: apps/web/public/static/locales/ru/common.json [hunk 15]
```diff
@@ -1614,8 +1614,6 @@
   "booking_confirmation_failed": "Не удалось подтвердить бронирование",
   "form_builder_field_already_exists": "Поле с таким именем уже существует",
   "form_builder_field_add_subtitle": "Настроить вопросы, задаваемые на странице бронирования",
-  "form_builder_system_field_cant_delete": "Это системное поле нельзя удалить.",
```
Comment (hariombalhara) [inline apps/web/public/static/locales/ru/common.json:1617]: unused translation now

File: packages/features/form-builder/Components.tsx [hunk 16]
```diff
@@ -325,8 +326,14 @@ export const Components: Record<BookingFieldType, Component> = {
                 })
               ) : (
                 // Show option itself as label because there is just one option
-                // TODO: Support asterisk for required fields
-                <Label>{options[0].label}</Label>
+                <>
+                  <Label>
+                    {options[0].label}
+                    {!readOnly && optionsInputs[options[0].value].required ? (
```
Comment (hariombalhara) [inline packages/features/form-builder/Components.tsx:332]: When showing the first option check that option's required status

**Review by hariombalhara — APPROVED** [review]

**Review by zomars — APPROVED** [review]
Code LGTM. Checking on why tests are failing.

**Pull request merged**